### DISCUSSION TITLE
Model testing functionality

### DISCUSSION THREAD

**Discussion by TjorvenDM** [deepjavalibrary/djl#1806] [Q&A]
Are there currently any classes in djl that support certain functions to test a model on your own dataset? (not for validating whilst training)
For example I want to see how well different models (densenet, resnet, ...) respond to my data in terms of accuracy, precision and recall to see which I should use best. Currently I'm using the folder/file names to check whether the prediction was right or not.

**Comment by zachgk** [comment]
You can try the `EasyTrain.evaluateDataset(...)`  and then call `trainer.getTrainingResult()` to get your data. You can then see the results with all of the Evaluators that you added to the trainer

**Reply by TjorvenDM** [reply]
Thank you for the answer, I've tried this but it doesn't seem to show me anything. I think I'm missing something in the config segment.
This is my code:
```
    public static void main(String[] args) throws ModelNotFoundException, MalformedModelException, IOException, TranslateException {
        Criteria<Image, DetectedObjects> criteria = Criteria.builder()
                .optApplication(Application.CV.OBJECT_DETECTION) 
                .setTypes(Image.class, DetectedObjects.class)
                .optFilter("backbone", "resnet50")
                .build();

        //load model
        ZooModel<Image, DetectedObjects> model = ModelZoo.loadModel(criteria);
        getResults(model);
        model.close();
    }
    public static void getResults(Model model) throws TranslateException, IOException {
        int batchSize = 32;
        Repository repo = Repository.newInstance("images", Paths.get("images"));
        ImageFolder dataset = ImageFolder.builder()
                .setRepository(repo)
                .addTransform(new Resize(100,100))
                .addTransform(new ToTensor())
                .setSampling(batchSize, true)
                .build();
        dataset.prepare();

        DefaultTrainingConfig config = new DefaultTrainingConfig(Loss.softmaxCrossEntropyLoss())
                .addEvaluator(new Accuracy());

        Trainer trainer = model.newTrainer(config);
        EasyTrain.evaluateDataset(trainer, dataset);
        System.out.println(trainer.getTrainingResult());
    }
```
This is my output: 
```
{
  "epoch": 0,
  "evaluations": {}
}
```
